### PR TITLE
feat(console): impl response statues ranges chart pie for V4 analytics

### PR DESCRIPTION
**Pull request by ThibaudAV**
WIP Note : change base when it's merge
## Issue

https://gravitee.atlassian.net/browse/APIM-XXX

## Description

A small description of what you did in that PR.

## Additional context

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-uktgnxfchf.chromatic.com)

### CODE REVIEW

File: gravitee-apim-repository/gravitee-apim-repository-api/src/main/java/io/gravitee/repository/log/v4/api/AnalyticsRepository.java [hunk 1]
```diff
@@ -29,4 +32,7 @@ public interface AnalyticsRepository {
     Optional<AverageAggregate> searchAverageMessagesPerRequest(QueryContext queryContext, AverageMessagesPerRequestQuery query);
 
     Optional<AverageAggregate> searchAverageConnectionDuration(QueryContext queryContext, AverageConnectionDurationQuery query);
+
+    @NonNull
```
Comment (ytvnr) [inline gravitee-apim-repository/gravitee-apim-repository-api/src/main/java/io/gravitee/repository/log/v4/api/AnalyticsRepository.java:36]: I think this annotation (coming from rxjava3 is useless)

File: gravitee-apim-repository/gravitee-apim-repository-elasticsearch/src/main/java/io/gravitee/repository/elasticsearch/v4/analytics/AnalyticsElasticsearchRepository.java [hunk 2]
```diff
@@ -75,4 +80,15 @@ public Optional<AverageAggregate> searchAverageConnectionDuration(QueryContext q
             .map(SearchAverageConnectionDurationResponseAdapter::adapt)
             .blockingGet();
     }
+
+    @Override
+    public @NonNull Optional<ResponseStatusRangeAggregate> searchResponseStatusRange(
```
Comment (ytvnr) [inline gravitee-apim-repository/gravitee-apim-repository-elasticsearch/src/main/java/io/gravitee/repository/elasticsearch/v4/analytics/AnalyticsElasticsearchRepository.java]: same remark for NonNull

File: gravitee-apim-console-webui/src/entities/management-api-v2/analytics/analyticsResponseStatusRanges.fixture.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { AnalyticsResponseStatusRanges } from './analyticsResponseStatusRanges';
+
+export const fakeAnalyticsResponseStatusRanges = (modifier?: Partial<AnalyticsResponseStatusRanges>): AnalyticsResponseStatusRanges => {
+  const base: AnalyticsResponseStatusRanges = {
+    ranges: {},
+    rangesByEntrypoint: {},
```
Comment (ytvnr) [inline gravitee-apim-console-webui/src/entities/management-api-v2/analytics/analyticsResponseStatusRanges.fixture.ts:22]: I think we modified the name in backend, don't forget to align the naming

File: gravitee-apim-console-webui/src/entities/management-api-v2/analytics/analyticsResponseStatusRanges.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+export interface AnalyticsResponseStatusRanges {
+  ranges: Record<string, number>;
+  rangesByEntrypoint: { [key: string]: Record<string, number> };
```
Comment (ytvnr) [inline gravitee-apim-console-webui/src/entities/management-api-v2/analytics/analyticsResponseStatusRanges.ts:18]: same

**Review by ytvnr — APPROVED** [review]

**Comment by dev-marek** [comment]
Replaced with PR: https://github.com/gravitee-io/gravitee-api-management/pull/7870

**Pull request closed without merging**